### PR TITLE
Pull loads and general load fixes

### LINKED ISSUE #120: Robot_Toolkit: Pull Loads to be implemented
Pull PointLoads has been implemented as a template.
Future loads to be prioritised and individual issues raised as appropriate

### LINKED ISSUE #125: Robot_Toolkit: Issue with pushing gravity loads with named groups
Trying to push gravity loads for all bars.Potential source of error: The gravity loads BHoM component will only accept a group of BHoMObjects not Bars, thus when pushed to Robot it is unable to assign the object correctly. Error message '1. Solution exception:Object reference not set to an instance of an object.'

### LINKED ISSUE #202: Robot_Toolkit: Enable local axis for Loads
Currently the project boolean and the axis enum is ignored when pushing most load types to Robot. Needs implementation.

### LINKED ISSUE #211: Robot_Toolkit:Pushing loads with groups creates empty loads
Push creates empty loads in Robot when either pushing loads with groups that don't exist in Robot or loads created with Create.Structure.Load that don't have bar ID's associated with the group.

### LINKED ISSUE #238: Robot_Toolkit: Gravity loading incorrect application of selfweight
#### Description:
Applying gravity loads works provided only g is required, and returns incorrect value for everything else. Also it only the load to the whole model regardless of the BHoM objects you provide it with

#### How to replicate:
Push a none 1.0 value vector.
Push loading only applied to part of the model.
Push vector which isn't aligned to the global axis.

#### Expected behaviour:
1. The applied load in load table should match the direction defined by the vector with the factor equalling the component in each of the global directions.
2. The load should only be applied to the objects defined.

### LINKED ISSUE #347: Robot_oM: The `Command` enum has no value
#### Broken rules:
An enum should always have at least one value for it to even make sense. 

#### Suggestions to restore compliance:
either delete that enum (it's not used anywhere) or complete its definition.

### LINKED ISSUE #348: Robot_Toolkit: Bar varying load is defined differently between BHoM and Robot
#### Description:

Second point in BHoM is measuring distance from end node, whilst in Robot both distances are measured from the start node.

### LINKED ISSUE #351: Robot_Toolkit: Panels normals are sometimes created flipped
#### Description:

Robot uses a different convention than BHoM. BHoM uses right-hand rule on external edges, Robot tries to align with world axes. This need to be checked during create/read

#### Test file(s):

https://burohappold.sharepoint.com/:f:/s/BHoM/EqbUk4v4hWlIpYaNoLKOKfkBisvSK_VdP3DVU518gLSLcQ?e=NIA3Kg

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### NOTE: Depends on 

 ### Issues addressed by this PR

Closes #120
Closes #238 
Closes #202 
Closes #125 
Closes #211 
Closes #347 
Closes #348 
Closes #351 

 

Genral comb through of loads, and adding capability of pulling all implemented load types available in Robot and BHoM.

Also, cleaning up redundant unused enums.

 ### Test files

Testing pushing through loads, comparing local, global projected:
https://burohappold.sharepoint.com/:u:/s/BHoM/EWJtw7n_SPdNtOdTVCOBkZsBccKJ2q5VdN4Dp8M1E5FwNQ?e=g5tnWE

Additional scripts:
https://burohappold.sharepoint.com/:f:/s/BHoM/ElNcfD7IGqVLrjrL1-zFvJ8BSOSYTZQF7Z3vea91Q5udJw?e=vOj54G

Panel normals:
https://burohappold.sharepoint.com/:f:/s/BHoM/EqbUk4v4hWlIpYaNoLKOKfkBisvSK_VdP3DVU518gLSLcQ?e=NIA3Kg

 ### Changelog

- Adding Pull functionality for load types not previously supported
- Fixing issue where groups with names did not get the loads applied to the element (fixed by disabling the assignment of loads to named groups for now)
- Fixing Gravity loads always applied to full model
- Make gravity load not split up into component to correctly show factors
- Fix wrong conversion for BarVaryingLoad between BHoM and Robot
- Enabling local and Global axis as well as projected for all loads where applicable.
- Fixing Panel normals sometimes read/created in flipped direction

 ### Additional comments

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@rwemay added link to some testfiles, one produced by @kThorsager when he added loads for Etabs, and one testing through the change to BarVaryingDistributed loads.

If you have some models or so to test on as well, that would be great!

File: Robot_Adapter/Convert/FromRobot/Loads/BarVaryingDistributedLoad.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            {
+                BarVaryingDistributedLoad clone = load.GetShallowClone() as BarVaryingDistributedLoad;
+                clone.DistanceFromB = lengthGroup.Key - clone.DistanceFromB; //Set distance from B to be from end node instead of start
+                clone.Objects = new BHoMGroup<Bar>() { Elements = lengthGroup.Value };
+                loads.Add(clone);
+                counter++;
+            }
+
+            if (counter > 1)
+                Engine.Reflection.Compute.RecordNote("Varying BarLoads in BHoM meassures distans from start for the first point and from end for the second point, whilst Robot measures only from start node. To accomodate this, load pulled from Robot as been split up in multiple loads.");
```
Comment (kThorsager) [inline Robot_Adapter/Convert/FromRobot/Loads/BarVaryingDistributedLoad.cs:91]:
```suggestion
                Engine.Reflection.Compute.RecordNote("Varying BarLoads in BHoM measures distance from start for the first point and from end for the second point, whilst Robot measures only from start node. To accommodate this, load pulled from Robot as been split up in multiple loads.");
```

Comment (kThorsager) [reply Robot_Adapter/Convert/FromRobot/Loads/BarVaryingDistributedLoad.cs:91]: I also don't get from the message nether why or how that difference lead to multiple loads (or how those multiple loads would look)? Can't we measure the Bar's length?

Checked the code after being confused and see that you mean that one load no longer encompasses all the bars it used to. Some clarification might be good

**Comment by IsakNaslundBh** [comment]
Added fix for Create/Read of panel normals, and added an additional test file for this. Good catch @kThorsager ! 👍

**Pull request merged**